### ISSUE TITLE
Order Creation: Accessibility → The Filter Countries search field should be marked as "Search Field" rather than "Text Field"

### ISSUE THREAD

**Issue by shiki** [woocommerce/woocommerce-ios#6077]
When navigating to the "Filter Countries" field using VoiceOver, it is dictated as a text field rather than a search field. Giving the right context that it's a search field can help clarify to the user how the current UI works. 

<img src="https://user-images.githubusercontent.com/198826/152437023-0a6b1941-9eff-4a45-9d2a-cd48a0996e4f.png" width="300">
